### DISCUSSION TITLE
Target an intent (or no intent) with a call

### DISCUSSION THREAD

**Discussion by Lex-Kent** [speechly/speechly#257] [General]
Hi all,

@teelisyys was kind enough to help me successfully create a set of intents which reliably understand the response to the question "Did I get your name right?" - which might be `heard_correctly_yes`, `heard_correctly_no` or `heard_correctly_unsure` : https://github.com/speechly/speechly/discussions/256#discussioncomment-4410386

However, I when ask a completely different question e.g. "Which of these options would you like to try?", and my user answers, say, "Option 1 please." then the model seems to think they are answering the "Did I get your name right?" question, and returns the intent it believes was being requested - `heard_correctly_no`.

How do I target either a specific set of intents, or specifically _no intent_ when I submit my user's speech? Or should I have a separate application for every single question?

**Comment by teelisyys** [comment]
Hi @Lex-Kent !

Unfortunately it is not possible to provide the expected intent along with the audio at this time. Our current suggested solution is indeed to use different apps for different prompts in use cases where strict separation of contexts required.

As a more general note: We have found that in many use cases it might be better to aim for a more open ended and less strictly question-answer based user experience, as we demonstrate in this [video](https://www.youtube.com/watch?v=XJ4BnEIiAjo). Of course prompting for specific data can also work well, especially in multimodal environments, as we demonstrate in this [video](https://www.youtube.com/watch?v=6GcgPcMOuQk).

**Comment by Lex-Kent** [comment]
Thanks @teelisyys, that's very helpful.

For now I'm having to use the question-answer methodology - the script has been developed by my client for a very specific financial app.

However, when these tricky parts of the app are finished, I am hoping to invoke my virtual assistant for general help tasks and in that role the conversation will (hopefully) be more like the videos you linked to.
